### PR TITLE
NMS-14889: Update PASSWORD-CLEAR parameter in docs and config examples

### PR DESCRIPTION
**Pull request by Bonrob2**
NMS-14889: Update PASSWORD-CLEAR parameter in docs and config examples: replaced "PASSWORD-CONFIG" with "PASSWORD_CONFIG" as noted in community chat.

### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14889

### CODE REVIEW

File: docs/modules/reference/pages/service-assurance/monitors/JmxMonitor.adoc [hunk 1]
```diff
@@ -31,7 +31,7 @@ The monitor offers the following functionalities:
 | from jmx-config.xml
 
 | factory
-| Set this to PASSWORD-CLEAR if credentials are required.
+| Set this to PASSWORD_CLEAR if credentials are required.
```
Comment (emarsh1995) [inline docs/modules/reference/pages/service-assurance/monitors/JmxMonitor.adoc:34]:
```suggestion
| Set this to `PASSWORD_CLEAR` if credentials are required.
```

**Review by emarsh1995 — COMMENTED** [review]
One minor change. Otherwise, docs LGTM. Thanks! :)

**Comment by mmahacek** [comment]
I had created https://issues.opennms.org/browse/NMS-14884 as a similar request.  Is there sign off from engineering as to what the correct expected value should be? While I think this is good to standardize the value, there will be older instances that may have either version and it may be worth updating the logic to handle either the dash and underscore versions interchangeably as to honor the intended value.

**Comment by Bonrob2** [comment]
> I had created https://issues.opennms.org/browse/NMS-14884 as a similar request. Is there sign off from engineering as to what the correct expected value should be? While I think this is good to standardize the value, there will be older instances that may have either version and it may be worth updating the logic to handle either the dash and underscore versions interchangeably as to honor the intended value.

No sign off from engineering. I'll close this PR and the ticket and we can move forward with yours once we have answers. Thanks.

**Comment by zatnas** [comment]
From my research down to the poller.log, I can see that by setting the value to `PASSWORD_CLEAR` instead of `PASSWORD-CLEAR` stops the WARN level logs.

Here is the log generator;
https://github.com/OpenNMS/opennms/blob/c3dec781a2e4aeee535c1dc6cd3bfb61d9200638/core/jmx/api/src/main/java/org/opennms/netmgt/jmx/connection/PasswordStrategy.java#L86-L92

Here is where the logic happens;
https://github.com/OpenNMS/opennms/blob/c3dec781a2e4aeee535c1dc6cd3bfb61d9200638/core/jmx/api/src/main/java/org/opennms/netmgt/jmx/connection/JmxConnectionConfig.java#L105-L113

And from the logs, it did hit log generator if the pollerd log level is set to WARN.

**Comment by mmahacek** [comment]
This PR and NMS ticket have been closed.  PR https://github.com/OpenNMS/opennms/pull/6237 has been opened to accept either `PASSWORD-CLEAR` or `PASSWORD_CLEAR` as a valid strategy.

**Pull request closed without merging**